src/Raven.Server/Documents/Indexes/Workers/HandleReferences.cs

```diff
@@ -212,6 +212,12 @@ private unsafe bool HandleItems(ActionType actionType, QueryOperationContext que
                                     hasChanges = true;
                                     inMemoryStats.UpdateLastEtag(lastEtag, isTombstone);
 
+                                    if (_index._indexStorage.LowerThanLastDatabaseEtagOnIndexCreation(referencedDocument.Etag))
```

Comment:
There is a mistake here, the variable should be called 'referencedItem'. It can be a compare exchange.

Do the following:
1. `referencedDocument` => `referencedItem`
2.  change that condition to: `if (ItemsAndReferencesAreUsingSameEtagPool && _index._indexStorage.LowerThanLastDatabaseEtagOnIndexCreation(referencedItem.Etag))`

Reply:
done

Reply:
Actually, this isn't true.
The `referencedItem` is always a `referencedDocument` since you can only have `LoadDocument`.